src/OpenApi/src/Services/OpenApiDocumentService.cs

```diff
@@ -68,12 +106,20 @@ private OpenApiPaths GetOpenApiPaths(HashSet<OpenApiTag> capturedTags)
         return paths;
     }
 
-    private static Dictionary<OperationType, OpenApiOperation> GetOperations(IGrouping<string?, ApiDescription> descriptions, HashSet<OpenApiTag> capturedTags)
+    private Dictionary<OperationType, OpenApiOperation> GetOperations(IGrouping<string?, ApiDescription> descriptions, HashSet<OpenApiTag> capturedTags)
     {
         var operations = new Dictionary<OperationType, OpenApiOperation>();
         foreach (var description in descriptions)
         {
-            operations[description.GetOperationType()] = GetOperation(description, capturedTags);
+            var operation = GetOperation(description, capturedTags);
+            operation.Extensions.Add(OpenApiConstants.DescriptionId, new OpenApiString(description.ActionDescriptor.Id));
+            OperationTransformerContextCache.TryAdd(description.ActionDescriptor.Id, new OpenApiOperationTransformerContext
```

Comment:
Why cache such a simple object?  Are we trying to avoid copies?

Reply:
Yes, we're reducing the overhead of creating a context object for each transformer which can get expensive fast. We're also saving the cost of having to do the lookup for the ActionDescriptor in the `DelegateOpenApiDocumentTransformer`.